target/scripts/helpers/accounts.sh

```diff
@@ -56,12 +59,27 @@ function _create_accounts
         _notify 'inf' "Creating user '${USER}' for domain '${DOMAIN}' with attributes '${USER_ATTRIBUTES}'"
       fi
 
-      echo "${LOGIN} ${DOMAIN}/${USER}/" >> /etc/postfix/vmailbox
+      local POSTFIX_VMAILBOX_LINE DOVECOT_USERDB_LINE
+
+      POSTFIX_VMAILBOX_LINE="${LOGIN} ${DOMAIN}/${USER}/"
+      if grep -qF "${POSTFIX_VMAILBOX_LINE}" /etc/postfix/vmailbox
+      then
+        _notify 'warn' "User '${USER}@${DOMAIN}' will not be added to '/etc/postfix/vmailbox' twice"
+      else
+        echo "${POSTFIX_VMAILBOX_LINE}" >>/etc/postfix/vmailbox
+      fi
+
       # Dovecot's userdb has the following format
       # user:password:uid:gid:(gecos):home:(shell):extra_fields
-      echo \
-        "${LOGIN}:${PASS}:5000:5000::/var/mail/${DOMAIN}/${USER}::${USER_ATTRIBUTES}" \
-        >>/etc/dovecot/userdb
+      DOVECOT_USERDB_LINE="${LOGIN}:${PASS}:5000:5000::/var/mail/${DOMAIN}/${USER}::${USER_ATTRIBUTES}"
+      if grep -qF "${DOVECOT_USERDB_LINE}" "${DOVECOT_USERDB_FILE}"
+      then
+        _notify 'warn' "Login '${LOGIN}' will not be added to '${DOVECOT_USERDB_FILE}' twice"
+      else
+        echo \
+          "${LOGIN}:${PASS}:5000:5000::/var/mail/${DOMAIN}/${USER}::${USER_ATTRIBUTES}" \
+          >>"${DOVECOT_USERDB_FILE}"
```

Comment:
You defined the line above now, any reason not to use it here too? Should be fine to collapse the echo line then?

```suggestion
        echo "${DOVECOT_USERDB_LINE}" >>"${DOVECOT_USERDB_FILE}"
```

Reply:
Yes, you're right.